### PR TITLE
use create time of replicasets because we don't have a good way to ge…

### PR DESCRIPTION
**Pull request by paulgmiller**
…t deployment update time

### CODE REVIEW

File: conditions.py [hunk 1]
```diff
@@ -15,26 +15,27 @@ def satisfy(namespace):
         return result
     return satisfy
 
-# no new deployment for certain hours, default to 24
-def InactiveDeploymentCondition(api_client_v1beta1, max_inactive_hours):
+
+def InactiveDeploymentCondition(api_client_v1, max_inactive_hours):
+    '''
+    no new replicaset or certain hours, default to 24
```
Comment (doubleyewdee) [inline conditions.py]: "for" not "or"

File: conditions.py [hunk 2]
```diff
@@ -15,26 +15,27 @@ def satisfy(namespace):
...
-            return (datetime.datetime.now(timezone) - deployment_condition.last_update_time) <= max_inactive_time
+        replica_sets = api_client.list_namespaced_replica_set(namespace.metadata.name)
 
-        def checkdeployment(d):
-            return any(is_active(c) for c in d.status.conditions if c.type == 'Progressing')
+        def is_active(replica_set):
+            created = replica_set.metadata.creation_timestamp
+            return (datetime.datetime.utcnow() - created) <= max_inactive_time
 
-        # at least one deployment is updated within max_inactive_days days, we consider this namespace active
```
Comment (doubleyewdee) [inline conditions.py]: It might be useful for the reader to know why we choose replicasets and not deployments. Up to you.

Comment (paulgmiller) [reply conditions.py]: yep...

File: conditions.py [hunk 3]
```diff
@@ -15,26 +15,27 @@ def satisfy(namespace):
...
-
-        def is_active(deployment_condition):
-            timezone = deployment_condition.last_update_time.tzinfo
-            return (datetime.datetime.now(timezone) - deployment_condition.last_update_time) <= max_inactive_time
+        replica_sets = api_client.list_namespaced_replica_set(namespace.metadata.name)
 
-        def checkdeployment(d):
-            return any(is_active(c) for c in d.status.conditions if c.type == 'Progressing')
+        def is_active(replica_set):
+            created = replica_set.metadata.creation_timestamp
```
Comment (paulgmiller) [inline conditions.py]: time to log created time.

**Pull request merged**